In [36]:
import pandas as pd

In [37]:
df = pd.read_csv("asthma_disease_data.csv")

Bu maʼlumotlar toʻplamida astma tashxisi qoʻyilgan 2392 nafar bemorning sogʻligʻi haqida keng maʼlumotlar mavjud boʻlib, ularning soni 5034 dan 7425 gacha boʻlgan noyob identifikatorlarga ega. U demografik, turmush tarzi omillari, atrof-muhit va allergiya omillari, kasallik tarixi, klinik oʻlchovlar, simptomlar va tashxis koʻrsatkichini oʻz ichiga oladi. Ushbu ma'lumotlar to'plami astma bilan bog'liq omillarni o'rganish, bashoratli modellarni ishlab chiqish va statistik tahlillarni amalga oshirishga intilayotgan tadqiqotchilar va ma'lumotlar olimlari (data scientistlar) uchun qimmatlidir.

In [38]:
pd.set_option("display.max_columns", None)

In [39]:
df.head().T

0          1          2          3          4
PatientID                    5034       5035       5036       5037       5038
Age                            63         26         57         40         61
Gender                          0          1          0          1          0
Ethnicity                       1          2          2          2          0
EducationLevel                  0          2          1          1          3
BMI                     15.848744  22.757042  18.395396  38.515278  19.283802
Smoking                         0          0          0          0          0
PhysicalActivity         0.894448   5.897329   6.739367   1.404503   4.604493
DietQuality              5.488696   6.341014   9.196237   5.826532   3.127048
SleepQuality             8.701003   5.153966   6.840647   4.253036   9.625799
PollutionExposure        7.388481   1.969838   1.460593   0.581905   0.980875
PollenExposure           2.855578   7.457665   1.448189   7.571845   3.049807
DustExposure             0.974339   6.584631   5.445799   3.965316   8.260605
PetAllergy                      1          0          0          0          0
FamilyHistoryAsthma             1          0          1          0          0
HistoryOfAllergies              0          1          1          0          0
Eczema                          0          0          0          0          0
HayFever                        0          0          1          1          1
GastroesophagealReflux          0          0          0          0          0
LungFunctionFEV1         1.369051   2.197767   1.698011   3.032037   3.470589
LungFunctionFVC          4.941206   1.702393   5.022553   2.300159   3.067944
Wheezing                        0          1          1          1          1
ShortnessOfBreath               0          0          1          0          1
ChestTightness                  1          0          1          1          1
Coughing                        0          1          0          1          0
NighttimeSymptoms               0          1          1          1          0
ExerciseInduced                 1          1          1          0          1
Diagnosis                       0          0          0          0          0
DoctorInCharge          Dr_Confid  Dr_Confid  Dr_Confid  Dr_Confid  Dr_Confid

In [40]:
df.columns

Index(['PatientID', 'Age', 'Gender', 'Ethnicity', 'EducationLevel', 'BMI',
       'Smoking', 'PhysicalActivity', 'DietQuality', 'SleepQuality',
       'PollutionExposure', 'PollenExposure', 'DustExposure', 'PetAllergy',
       'FamilyHistoryAsthma', 'HistoryOfAllergies', 'Eczema', 'HayFever',
       'GastroesophagealReflux', 'LungFunctionFEV1', 'LungFunctionFVC',
       'Wheezing', 'ShortnessOfBreath', 'ChestTightness', 'Coughing',
       'NighttimeSymptoms', 'ExerciseInduced', 'Diagnosis', 'DoctorInCharge'],
      dtype='object')

ustunlarning nomlari: 
bemorning ID si, 
yoshi, 
jinsi, 
etnik kelib chiqishi, 
ta'lim darajasi, 
BMI, chekadimi, 
jismoniy faolligi, 
dietasinig sifati, 
uyqu sifati, 
ifloslanish ta'siri, 
Polen ta'sir qilishi(allegiya turi), 
chang ta'sir qilishi, 
uy hayvonlariga allergiyasi, 
oila tarixida astma, 
allergiya tarixi, 
ekzema(ter kasalligi turi), 
pichan isitmasi(allergiya turi), 
gastroezofegal reflyuks(oshqozondagi ovqatni doimo og'ziga kelishi xronik kasallik), 
o'pka funktsiyasi fev 1(1 sekundda qancha hajm havoni o'pkadan chiqishi), 
o'pka funktsiyasi fvc(kuch bilan qancha hajm havoni o'pkadan chiqishi), 
xirillash, 
nafas qisilishi, 
ko'krak qafasi qisilishi,
yo'tal, 
tungi vaqt belgilari, 
jismoniy mashqlar, 
tashxis, 
mas'ul shifokor

# Bemor haqida ma'lumot
## Bemor ID
### PatientID: Har bir bemorga tayinlangan noyob identifikator (5034 dan 7425 gacha).

## Demografiya
### Yoshi: Bemorlarning yoshi 5 yoshdan 80 yoshgacha.
### Jins: Bemorlarning jinsi, bu erda 0 erkak va 1 ayolni bildiradi.
### Etnik kelib chiqishi: Bemorlarning etnik kelib chiqishi quyidagicha kodlangan:
#### 0: Kavkaz
#### 1: afro-amerikalik
#### 2: Osiyo
#### 3: Boshqa
### Ta'lim darajasi: Bemorning ta'lim darajasi, quyidagi kodlangan:
#### 0: Yo'q
#### 1: Oliy maktab
#### 2: Bakalavr darajasi
#### 3: Yuqori
## Turmush tarzi omillari
### BMI: bemorlarning tana massasi indeksi 15 dan 40 gacha.
### Chekish: Chekish holati, bu erda 0 "yo'q" va "1" "ha" degan ma'noni anglatadi.
### Jismoniy faollik: 0 dan 10 gacha bo'lgan soatlarda haftalik jismoniy faollik.
### DietQuality: dietaning sifati 0 dan 10 gacha.
### SleepQuality: 4 dan 10 gacha bo'lgan uyqu sifati reytingi.
## Atrof-muhit omillari va allergiya
### PollutionExposure: Ifloslanishga ta'sir qilish, 0 dan 10 gacha ball.
### Pollen ta'siri: 0 dan 10 gacha bo'lgan polen ta'siri.
### Changga ta'sir qilish: 0 dan 10 gacha baholangan changga ta'sir qilish.
### Uy hayvonlari allergiyasi: uy hayvonlari allergiyasi holati, bu erda 0 "yo'q" va "1" "ha" degan ma'noni anglatadi.
## Kasallik tarixi
### FamilyHistoryAstma: Oilaviy astma tarixi, bu erda 0 "yo'q" va "1" "ha" degan ma'noni anglatadi.
### HistoryOfAllergies : Allergiya tarixi, bu erda 0 yo'q va 1 "ha" degan ma'noni anglatadi.
### Egzema: ekzemaning mavjudligi, bu erda 0 yo'q va 1 "ha" degan ma'noni anglatadi.
### HayFever: pichan isitmasi bor, bu erda 0 yo'q va 1 ha degani.
### Gastroezofagial reflyuks: gastroezofagial reflyuksning mavjudligi, bu erda 0 "yo'q" va 1 "ha" degan ma'noni anglatadi.
## Klinik o'lchovlar
### LungFunctionFEV1: 1 soniyada majburiy ekspiratsiya hajmi (FEV1) 1,0 dan 4,0 litrgacha.
### LungFunctionFVC: majburiy hayotiy sig'im (FVC) 1,5 dan 6,0 litrgacha.
## Alomatlar
### Wheezing: xirillashning mavjudligi, bu erda 0 "yo'q" va 1 "ha" degan ma'noni anglatadi.
### ShortnessOfBreath: nafas qisilishining mavjudligi, bu erda 0 "yo'q" va 1 "ha" degan ma'noni anglatadi.
### Ko'krak qafasining siqilishi: ko'krak qafasidagi siqilishning mavjudligi, bu erda 0 "yo'q" va 1 "ha" degan ma'noni anglatadi.
### Yo'tal: yo'talning mavjudligi, bu erda 0 "yo'q" va 1 "ha" degan ma'noni anglatadi.
### Tungi simptomlar: tungi simptomlarning mavjudligi, bu erda 0 yo'q va 1 ha degani.
### Ishga tushirilgan mashqlar: Jismoniy mashqlar tufayli alomatlar paydo bo'ladi, bu erda 0 "Yo'q" va 1 "Ha" degan ma'noni anglatadi.
## Diagnostik ma'lumotlar
### Tashxis: Nafas tashxisi holati, bu erda 0 "yo'q" va 1 "ha" degan ma'noni anglatadi.
## Maxfiy ma'lumotlar
### DoctorInCharge : Ushbu ustunda barcha bemorlar uchun "Dr_Confid" qiymatiga ega bo'lgan davolovchi shifokor haqidagi maxfiy ma'lumotlar mavjud.

# Demografic

In [41]:
df_demografic = df[["PatientID","Age","Gender","Ethnicity","EducationLevel","Diagnosis"]]
df_demografic

PatientID  Age  Gender  Ethnicity  EducationLevel  Diagnosis
0          5034   63       0          1               0          0
1          5035   26       1          2               2          0
2          5036   57       0          2               1          0
3          5037   40       1          2               1          0
4          5038   61       0          0               3          0
...         ...  ...     ...        ...             ...        ...
2387       7421   43       1          0               2          1
2388       7422   18       1          0               1          1
2389       7423   54       0          3               2          1
2390       7424   46       1          0               2          0
2391       7425   26       1          0               0          0

[2392 rows x 6 columns]

In [42]:
df_demografic.describe()

PatientID          Age       Gender    Ethnicity  EducationLevel  \
count  2392.000000  2392.000000  2392.000000  2392.000000     2392.000000   
mean   6229.500000    42.137960     0.493311     0.669732        1.307274   
std     690.655244    21.606655     0.500060     0.986120        0.898242   
min    5034.000000     5.000000     0.000000     0.000000        0.000000   
25%    5631.750000    23.000000     0.000000     0.000000        1.000000   
50%    6229.500000    42.000000     0.000000     0.000000        1.000000   
75%    6827.250000    61.000000     1.000000     1.000000        2.000000   
max    7425.000000    79.000000     1.000000     3.000000        3.000000   

         Diagnosis  
count  2392.000000  
mean      0.051839  
std       0.221749  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       0.000000  
max       1.000000

In [43]:
df_demografic[["Age","Gender","Ethnicity","EducationLevel","Diagnosis"]].mode()

Age  Gender  Ethnicity  EducationLevel  Diagnosis
0   51       0          0               1          0

In [44]:
def pct_n(column, n):
    return column.quantile(n/100)
df_demografic["EducationLevel"].agg(pct_n, n=19)

np.float64(0.0)

bemorlarning o'rtacha yoshi 42da, eng kichik yosh - 5, kattasi - 79, modasi - 51 yoshdir, ayollar va erkaklar deyarli teng, etnic tuzilishida kafkazlar ko'proqdir, maktab bitiruvchilari ko'p qismini tashkil qiladi, ulardan keyin bakalavr bitirganlardir, 19% - hech qayerdi bitirmaganlar, 10 - magistratura va undan baland

In [45]:
df_demografic[["Age","EducationLevel", "Diagnosis"]].cumsum()

Age  EducationLevel  Diagnosis
0         63               0          0
1         89               2          0
2        146               3          0
3        186               4          0
4        247               7          0
...      ...             ...        ...
2387  100650            3122        122
2388  100668            3123        123
2389  100722            3125        124
2390  100768            3127        124
2391  100794            3127        124

[2392 rows x 3 columns]

In [46]:
df_demografic['Diagnosis'].describe()*100

count    239200.000000
mean          5.183946
std          22.174916
min           0.000000
25%           0.000000
50%           0.000000
75%           0.000000
max         100.000000
Name: Diagnosis, dtype: float64

Astma bilan kasallanganlar jami 124 ta ekan 2392 tadan bu o'rtacha 5.18 % dir

In [47]:
df_demografic[["Age"]].cummax()

Age
0      63
1      63
2      63
3      63
4      63
...   ...
2387   79
2388   79
2389   79
2390   79
2391   79

[2392 rows x 1 columns]

In [48]:
df_demografic["Age"].cummin()

0       63
1       26
2       26
3       26
4       26
        ..
2387     5
2388     5
2389     5
2390     5
2391     5
Name: Age, Length: 2392, dtype: int64

jadvalar ko'pligi sababli izoh berish qiyin bularga(cummax, cummin), ammo jadvaldagi max yoki minimumni o'zida saqlab yuradi, bunda min yoki max ni kuzatish oson

In [49]:
df_demografic.drop_duplicates(subset="Age")

PatientID  Age  Gender  Ethnicity  EducationLevel  Diagnosis
0         5034   63       0          1               0          0
1         5035   26       1          2               2          0
2         5036   57       0          2               1          0
3         5037   40       1          2               1          0
4         5038   61       0          0               3          0
..         ...  ...     ...        ...             ...        ...
199       5233   78       1          0               1          0
200       5234   24       0          3               1          0
220       5254   11       0          0               1          0
349       5383    6       1          0               1          0
398       5432   59       0          0               1          0

[75 rows x 6 columns]

Esimizda yoshlar 5 dan 79 gacha edi demak, bu yerda 75 qator bo'lsa demak bizda hamma yoshdagi bemorlarimiz bor ekan

In [50]:
df_demografic.drop_duplicates(subset=["Age","Gender","Ethnicity","Diagnosis"]).shape

(635, 6)

In [51]:
df_demografic.drop_duplicates(subset=["Age","Gender","Ethnicity"]).shape

(534, 6)

In [52]:
df_demografic.drop_duplicates(subset=["Age","Gender","Diagnosis"]).shape

(236, 6)

In [53]:
df_demografic.drop_duplicates(subset=["Age","Ethnicity","Diagnosis"]).shape

(389, 6)

In [54]:
df_demografic.drop_duplicates(subset=["Gender","Ethnicity","Diagnosis"]).shape

(16, 6)

In [55]:
df_demografic.drop_duplicates(subset=["Age","Gender"]).shape

(150, 6)

In [56]:
df_demografic.drop_duplicates(subset=["Age","Ethnicity"]).shape

(296, 6)

In [57]:
df_demografic.drop_duplicates(subset=["Gender","Ethnicity"]).shape

(8, 6)

In [58]:
df_demografic.drop_duplicates(subset=["Age","Diagnosis"]).shape

(134, 6)

In [59]:
df_demografic.drop_duplicates(subset=["Ethnicity","Diagnosis"]).shape

(8, 6)

yuqoridagilarni ham analiz qilsak etnik, yosh, diagnoz, jinsi bo'yicha

In [60]:
df_demografic.groupby("Gender")[['Age',"Diagnosis","EducationLevel"]].mean()

Age  Diagnosis  EducationLevel
Gender                                      
0       42.206271   0.051155        1.330858
1       42.067797   0.052542        1.283051

bemorlar orasida ayollarni ham erkaklarni ham o'rtacha yoshi 42 ekan, Ikkalasida ham kasal bo'lish foizi teng ekan

In [61]:
df_demografic.groupby("Diagnosis")[['Age',"Gender","EducationLevel"]].mean()

Age    Gender  EducationLevel
Diagnosis                                     
0          42.214286  0.492945        1.305556
1          40.741935  0.500000        1.338710

Haqiqatdan ham kasal bo'lganlar erkaklar - 42, ayollar - 41 yosh atrofida

In [62]:
df_demografic.groupby("EducationLevel")[['Age',"Gender","Diagnosis"]].mean()

Age    Gender  Diagnosis
EducationLevel                                
0               41.451883  0.512552   0.052301
1               43.049303  0.491961   0.054662
2               41.596796  0.497997   0.038718
3               41.633621  0.443966   0.081897

bilimiz kam bo'lsa yoki juda ham ko'p bo'lsa kasal bo'lasiz ekan, yaxshisi bakalavr darajasi, tinchkina uzoqroq yashaysiz

In [63]:
df_demografic.groupby("Diagnosis")["Age"].agg([min, max, sum])

C:\Users\Rudy\AppData\Local\Temp\ipykernel_1588\2999327215.py:1: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  df_demografic.groupby("Diagnosis")["Age"].agg([min, max, sum])
C:\Users\Rudy\AppData\Local\Temp\ipykernel_1588\2999327215.py:1: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  df_demografic.groupby("Diagnosis")["Age"].agg([min, max, sum])
C:\Users\Rudy\AppData\Local\Temp\ipykernel_1588\2999327215.py:1: FutureWarning: The provided callable <built-in function sum> is currently using SeriesGroupBy.sum. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the stri

min  max    sum
Diagnosis                 
0            5   79  95742
1            5   79   5052

In [64]:
df_demografic.groupby("Age")["Diagnosis"].agg([min, max, sum]).head(20)

C:\Users\Rudy\AppData\Local\Temp\ipykernel_1588\2280454528.py:1: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  df_demografic.groupby("Age")["Diagnosis"].agg([min, max, sum]).head(20)
C:\Users\Rudy\AppData\Local\Temp\ipykernel_1588\2280454528.py:1: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  df_demografic.groupby("Age")["Diagnosis"].agg([min, max, sum]).head(20)
C:\Users\Rudy\AppData\Local\Temp\ipykernel_1588\2280454528.py:1: FutureWarning: The provided callable <built-in function sum> is currently using SeriesGroupBy.sum. In a future version of pandas, the provided callable will be used directly. To keep current beha

min  max  sum
Age               
5      0    1    5
6      0    1    2
7      0    1    3
8      0    0    0
9      0    1    2
10     0    0    0
11     0    0    0
12     0    1    4
13     0    1    2
14     0    1    1
15     0    1    1
16     0    1    4
17     0    1    1
18     0    1    3
19     0    1    1
20     0    1    2
21     0    1    1
22     0    1    1
23     0    1    3
24     0    1    2

juda ham yaxshi jadval. Yoshlar bo'yicha kasallar sonini bilish mumkin

In [65]:
df_age_diagnosis = df_demografic.groupby("Age")["Diagnosis"].agg([min, max, sum])
df_age_diagnosis_filtered = df_age_diagnosis[~(df_age_diagnosis[['min', 'max', 'sum']] == 0).all(axis=1)]
df_age_diagnosis_filtered.head()

C:\Users\Rudy\AppData\Local\Temp\ipykernel_1588\388686786.py:1: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  df_age_diagnosis = df_demografic.groupby("Age")["Diagnosis"].agg([min, max, sum])
C:\Users\Rudy\AppData\Local\Temp\ipykernel_1588\388686786.py:1: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  df_age_diagnosis = df_demografic.groupby("Age")["Diagnosis"].agg([min, max, sum])
C:\Users\Rudy\AppData\Local\Temp\ipykernel_1588\388686786.py:1: FutureWarning: The provided callable <built-in function sum> is currently using SeriesGroupBy.sum. In a future version of pandas, the provided callable will be used directly. To 

min  max  sum
Age               
5      0    1    5
6      0    1    2
7      0    1    3
9      0    1    2
12     0    1    4

In [66]:
df_age_diagnosis_filtered.shape

(59, 3)

In [67]:
df_demografic.groupby(["EducationLevel", "Gender"])[['Age',"Diagnosis"]].mean()

Age  Diagnosis
EducationLevel Gender                      
0              0       42.137339   0.047210
               1       40.800000   0.057143
1              0       42.864979   0.061181
               1       43.239651   0.047930
2              0       42.204787   0.031915
               1       40.983914   0.045576
3              0       39.914729   0.077519
               1       43.786408   0.087379

In [68]:
df_demografic.groupby(["EducationLevel", "Gender","Diagnosis"])['Age'].mean()

EducationLevel  Gender  Diagnosis
0               0       0            41.743243
                        1            50.090909
                1       0            41.125541
                        1            35.428571
1               0       0            43.092135
                        1            39.379310
                1       0            43.169336
                        1            44.636364
2               0       0            42.629121
                        1            29.333333
                1       0            40.814607
                        1            44.529412
3               0       0            40.537815
                        1            32.500000
                1       0            43.223404
                        1            49.666667
Name: Age, dtype: float64

Series to DataFrame

In [69]:
df_demografic.groupby(["EducationLevel", "Gender","Diagnosis"])['Age'].mean().reset_index()

EducationLevel  Gender  Diagnosis        Age
0                0       0          0  41.743243
1                0       0          1  50.090909
2                0       1          0  41.125541
3                0       1          1  35.428571
4                1       0          0  43.092135
5                1       0          1  39.379310
6                1       1          0  43.169336
7                1       1          1  44.636364
8                2       0          0  42.629121
9                2       0          1  29.333333
10               2       1          0  40.814607
11               2       1          1  44.529412
12               3       0          0  40.537815
13               3       0          1  32.500000
14               3       1          0  43.223404
15               3       1          1  49.666667

Tugadi. Asosiy ko'rsatkichlar bilan ishlandi